ccm_web/server/src/lti/lti.service.ts

```diff
@@ -66,6 +67,17 @@ export class LTIService implements BeforeApplicationShutdown {
       const roles = customLTIVariables.roles as string
       const isRootAdmin = customLTIVariables.is_root_account_admin as boolean
 
+      // check whether the user has at least one of the allowed LTI user roles
+      // otherwise block tool access and show error message
+      const rolesLTI = roles.length > 0 ? roles.split(',') : []
+      const ltiAllowedRoles = rolesLTI.filter(
+        x => Object.entries(LTIEnrollmentTypes).find(
```

Comment:
Nit-pick: You could move `Object.entries(LTIEnrollmentTypes)` above  the `filter` and save the result in a `const`. I'm guessing the code would repeat the `entries` creation this way for each element in `rolesLTI`, which is probably a trivial calculation but not one you need to repeat.

Reply:
updated the code block as suggested.